src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1778,6 +1864,23 @@ def _configure_security_policy_storage_params(arg_ctx):
                    options_list=['--maint-config-id', '-m'],
                    help='Assign maintenance configuration to this managed instance.')
 
+        c.argument('enable_ad_only_auth',
```

Comment:
recommend to use `store_true` if only true is accpected

Reply:
i have tested and recorded tests with this, out of scope to modify now, api accepts true, false and null, so it's a general boolean

Reply:
The current definition is not boolean type. It will accept any string.